airbyte-integrations/connectors/source-snowflake/unit_tests/integration/request_builder.py

```diff
@@ -92,16 +97,46 @@ def _build_query_params(self, is_get: bool = False):
                 query_params = {}
         return query_params
 
+    def get_state_sql_condition(self):
+        if self._state is None:
+            return None
+
+        state_sql_condition = None
+        current_state_value = self._state[self._cursor_field]
+
+        if self._cursor_generic_type == "timestamp_with_timezone":
```

Comment:
I have created a statement builder to handle this

Reply:
I'm not sure I get the change. The Snowflake request builder still expose Airbyte concepts that have nothing to do with Snowflake [here](https://github.com/artefactory/airbyte/blob/9b6f9b93bce43cb23a557a5f9acd8b6f2e32d7cc/airbyte-integrations/connectors/source-snowflake/unit_tests/integration/request_builder.py#L78-L84. It also duplicates code path logic in the `build` method. 

Here is what I would expect as an interface for a SnowflakeRequestBuilder:
```
class SnowflakeRequestBuilder:

    def with_statement(self, statement: str) -> "SnowflakeRequestBuilder":
        <...>

    def with_role(self, role: str) -> "SnowflakeRequestBuilder":
        <...>

    def with_warehouse(self, warehouse: str) -> "SnowflakeRequestBuilder":
        <...>

    def with_database(self, database: str) -> "SnowflakeRequestBuilder":
        <...>

    def with_timeout(self, timeout: str) -> "SnowflakeRequestBuilder":
        <...>

    def with_request_id(self, timeout: str) -> "SnowflakeRequestBuilder":
        <...>

    def with_pagination(self, timeout: str) -> "SnowflakeRequestBuilder":
        <...>

    def build(self) -> HttpRequest:
        <...>
```

The rest is Airbyte domain and should not be mixed here because it will force us to duplicate prod code path logic to the builder which makes them test that we copy/pasted properly and not that we've set the expectations properly

Reply:
It will be corrected in the current latest [PR](https://github.com/artefactory/airbyte/pull/45) because the code was refactored on that branch meaning less changes on the code.